### DISCUSSION TITLE
Search by key

### DISCUSSION THREAD

**Discussion by chaptersix** [provectus/kafka-ui#2428] [Q&A] [answered]
How do I search a topic's messages by key? the only searches that I've seen look at message content only.

**Comment by Haarolean** [comment, marked as answer]
Considering current implementation the search should work fine with both key and value.
Also, you can try out smart filters for more granular search (#127):
<img width="523" alt="image" src="https://user-images.githubusercontent.com/1494347/184602319-8de323b2-539b-4dc2-b1da-1687dd1036b6.png">
